core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/TupleCodec.java

```diff
@@ -186,15 +187,25 @@ public TupleValue parse(@Nullable String value) {
       } catch (IllegalArgumentException e) {
         throw new IllegalArgumentException(
             String.format(
-                "Cannot parse tuple value from \"%s\", invalid CQL value at character %d",
-                value, position),
+                "Cannot parse tuple value from \"%s\", invalid CQL value at character %d: %s",
+                value, position, e.getMessage()),
             e);
       }
 
       String fieldValue = value.substring(position, n);
       DataType elementType = cqlType.getComponentTypes().get(i);
       TypeCodec<Object> codec = registry.codecFor(elementType);
-      tuple = tuple.set(i, codec.parse(fieldValue), codec);
+      Object parsed;
+      try {
+        parsed = codec.parse(fieldValue);
+      } catch (Exception e) {
+        throw new IllegalArgumentException(
+            String.format(
+                "Cannot parse tuple value from \"%s\", invalid CQL value at character %d: %s",
+                value, position, e.getMessage()),
+            e);
```

Comment:
Since we also know the field index, it would be nice to include it in the message as well:
> invalid CQL value at field 0 (character 1)

Same with the field name for UDTs.

Reply:
Good idea. Done. But that made the whole changeset more involved than I initially thought it would be. Hope that's ok.